src/java/htsjdk/variant/vcf/AbstractVCFCodec.java

```diff
@@ -752,13 +753,15 @@ private final static boolean isVCFStream(final InputStream stream, final String
         return new LazyGenotypesContext.LazyData(genotypes, header.getSampleNamesInOrder(), header.getSampleNameToOffset());
     }
 
-    private final String[] INT_DECODE_ARRAY = new String[10000];
     private final int[] decodeInts(final String string) {
```

Comment:
I prefer to make methods that do not depend on any instance variables as static, in part to implicitly convey that fact.

At least in theory, invocation of static methods could be slightly faster because they obviate a null check for the instance. I'm not sure how the JVM implements this in practice.